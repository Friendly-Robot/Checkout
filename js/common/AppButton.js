/**
 * @flow
 */

"use strict";

import React from "react";
import AppColors from "./AppColors";
import { View, StyleSheet, TouchableOpacity, Image, Alert } from "react-native";
import { Text } from "./AppText";

const BUTTON_HEIGHT = 52,
  BUTTON_HEIGHT_SM = 32;

class AppButton extends React.Component {
  props: {
    theme:
      | "blue"
      | "white"
      | "bordered",
    type: "default" | "round" | "small",
    opacity: number,
    icon?: number,
    caption?: string,
    captionStyle?: object,
    hitSlop?: number,
    style?: any,
    fontSize?: number,
    onPress: () => mixed
  };

  static defaultProps = {
    hitSlop: 0,
    opacity: 1,
    theme: "white"
  };

  static height = BUTTON_HEIGHT;

  render() {
    const { caption, captionStyle, icon, fontSize, opacity } = this.props;
    const { buttonTheme, iconTheme, captionTheme } = this.getTheme();
    const { containerType, buttonType, iconType, captionType } = this.getType();
    const hitSlop = this.getHitSlop();

    let iconImage;
    if (icon) {
      iconImage = (
        <Image source={icon} style={[styles.icon, iconTheme, iconType]} />
      );
    }

    let fontSizeOverride;
    if (fontSize) {
      fontSizeOverride = { fontSize };
    }

    const content = (
      <View style={[styles.button, buttonTheme, buttonType, { opacity }]}>
        <Text
          style={[styles.caption, captionTheme, captionType, fontSizeOverride, captionStyle]}
        >
          {caption}
        </Text>
        {iconImage}
      </View>
    );

    if (this.props.onPress) {
      return (
        <TouchableOpacity
          accessibilityTraits="button"
          activeOpacity={0.5}
          onPress={this.props.onPress}
          hitSlop={hitSlop}
          style={[styles.container, containerType, this.props.style]}
        >
          {content}
        </TouchableOpacity>
      );
    } else {
      return (
        <View style={[styles.container, containerType, this.props.style]}>
          {content}
        </View>
      );
    }
  }

  getTheme() {
    const { theme } = this.props;
    let buttonTheme, iconTheme, captionTheme;
    if (theme === "bordered") {
      buttonTheme = {
        backgroundColor: "transparent",
        borderColor: AppColors.primary,
        borderWidth: 1
      };
      iconTheme = { tintColor: AppColors.primary };
      captionTheme = { color: AppColors.primary };
    } else if (theme === "blue") {
      buttonTheme = { backgroundColor: AppColors.blue };
      iconTheme = { tintColor: AppColors.white };
      captionTheme = { color: AppColors.white };
    } else if (theme === "white") {
      buttonTheme = { backgroundColor: AppColors.white };
      iconTheme = { tintColor: AppColors.primary };
      captionTheme = { color: AppColors.primary };
    } else if (theme === "none") {
      buttonTheme = { backgroundColor: "transparent", padding: 0 };
      iconTheme = { tintColor: AppColors.primary };
      captionTheme = { color: AppColors.primary };
    } else {
      buttonTheme = { backgroundColor: AppColors.primary };
      iconTheme = { tintColor: AppColors.white };
      captionTheme = { color: AppColors.white };
    }

    return { buttonTheme, iconTheme, captionTheme };
  }

  getType() {
    const { type } = this.props;
    let containerType, buttonType, iconType, captionType;

    if (type === "round") {
      buttonType = { width: BUTTON_HEIGHT, paddingHorizontal: 0 };
      iconType = { marginRight: 0 };
      captionType = { fontSize: 13 };
    } else if (type === "small") {
      containerType = { height: BUTTON_HEIGHT_SM };
      buttonType = { paddingHorizontal: 15 };
      iconType = { marginRight: 0 };
      captionType = { fontSize: 13 };
    } else if (type === "none") {
      captionType = { textAlign: "left" };
      buttonType = { justifyContent: "flex-start", paddingHorizontal: 0, borderRadius: 0 };
    } else {
      // defaults
    }

    return { containerType, buttonType, iconType, captionType };
  }

  getHitSlop() {
    const { hitSlop } = this.props;
    return { top: hitSlop, right: hitSlop, bottom: hitSlop, left: hitSlop };
  }
}

const styles = StyleSheet.create({
  container: {
    height: BUTTON_HEIGHT
  },
  button: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 30,
    borderRadius: BUTTON_HEIGHT / 2
  },
  buttonRound: {
    width: BUTTON_HEIGHT,
    paddingHorizontal: 0
  },
  icon: {
    marginRight: 12
  },
  caption: {
    fontSize: 15,
    textAlign: "center"
  }
});

export default AppButton;
