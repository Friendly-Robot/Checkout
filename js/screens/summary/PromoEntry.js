/**
 * @flow
 */

"use strict";

import React, { useState } from "react";
import { StyleSheet, View } from "react-native";
import AppColors from "../../common/AppColors";
import { Heading3 } from "../../common/AppText";
import AppInput from "../../common/AppInput";
import AppButton from "../../common/AppButton";
import DualItemRow from "../../common/DualItemRow";

type Props = {
  handlePromoSubmit: function
};
const PromoEntry = ({
  handlePromoSubmit = () => {}
}: Props) => {

  const [promo, setPromo] = useState("");

  return (
    <View style={styles.wrapper}>
      <Heading3 style={styles.label}>Promo code</Heading3>
      <DualItemRow itemRightStyle={styles.buttonWrapper}>
        <AppInput
          autoCapitalize={"characters"}
          multiline={false}
          maxLength={50}
          inputColor={AppColors.coolGray}
          textColor={AppColors.black}
          onChange={setPromo}
          type={"box"}
          underlineColor={AppColors.white}
        />
        <AppButton
          caption={"Apply"}
          hitSlop={5}
          onPress={() => handlePromoSubmit(promo)}
          theme={"bordered"}
          type={"small"}
        /> 
      </DualItemRow>
    </View>
  )
}

const styles = StyleSheet.create({
  label: {
    color: AppColors.lightGray,
    marginLeft: 10
  },
  buttonWrapper: {
    alignItems: "flex-start",
    flex: 0.3,
    marginLeft: 15
  },
  wrapper: {
    marginVertical: 10
  }
});

export default PromoEntry;
