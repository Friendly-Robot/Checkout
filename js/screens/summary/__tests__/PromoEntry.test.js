/**
 * @format
 */

import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import AppButton from "../../../common/AppButton";
import PromoEntry from "../PromoEntry";


describe("<PromoEntry />", () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      handlePromoSubmit: () => {},
    };

    wrapper = shallow(<PromoEntry {...props} />);
  })

  it("should be defined", () => {
    expect(PromoEntry).toBeDefined();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<PromoEntry {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle press action for ", () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('Apply button pressed');

    wrapper = shallow(<PromoEntry handlePromoSubmit={onPressEvent} />)
    const button = wrapper.find(AppButton);
    button.props().onPress();
    expect(onPressEvent.mock.calls.length).toBe(1);
  });
});
