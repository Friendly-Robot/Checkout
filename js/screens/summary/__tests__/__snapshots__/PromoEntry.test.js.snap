// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoEntry /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(3, 3, 3, 1)",
          "fontSize": 17,
        },
        Object {
          "color": "rgba(208, 208, 208, 1)",
          "marginLeft": 10,
        },
      ]
    }
  >
    Promo code
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "marginRight": 10,
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "justifyContent": "center",
              "marginVertical": 5,
              "paddingHorizontal": 5,
              "paddingVertical": 5,
            },
            Object {
              "borderColor": "rgba(208, 208, 208, 1)",
              "borderRadius": 1,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="characters"
          maxLength={50}
          multiline={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "rgba(0, 114, 206, 1)",
                "fontSize": 17,
              },
              Object {
                "borderColor": undefined,
                "color": "rgba(3, 3, 3, 1)",
              },
            ]
          }
          textAlign="left"
          underlineColorAndroid="rgba(255, 255, 255, 1)"
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "marginLeft": 10,
          },
          Object {
            "alignItems": "flex-start",
            "flex": 0.3,
            "marginLeft": 15,
          },
        ]
      }
    >
      <View
        accessible={true}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 32,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 26,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "borderColor": "rgba(0, 114, 206, 1)",
                "borderWidth": 1,
              },
              Object {
                "paddingHorizontal": 15,
              },
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(3, 3, 3, 1)",
                },
                Array [
                  Object {
                    "fontSize": 15,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 114, 206, 1)",
                  },
                  Object {
                    "fontSize": 13,
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Apply
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
